chpl/chpl-service/src/main/java/gov/healthit/chpl/report/criteriaattribute/StandardReportDao.java

```diff
@@ -0,0 +1,108 @@
+                + "CertificationResultEntity cr, "
+                + "CertifiedProductDetailsEntity cpd, "
+                + "CertificationResultStandardEntity crs, "
+                + "StandardEntity s "
+                + "WHERE cc.id = cr.certificationCriterionId "
+                + "AND cr.certifiedProductId = cpd.id "
+                + "AND cr.id = crs.certificationResultId "
+                + "AND crs.standard.id = s.id "
+                + "AND cpd.certificationStatusId IN (1,6,7) "
+                + "AND (cc.endDay is null OR cc.endDay > CURRENT_DATE()) "
+                + "AND cc.deleted = false "
+                + "AND cr.deleted = false "
+                + "AND crs.deleted = false "
+                + "AND cpd.deleted = false "
+                + "AND s.deleted = false "
+                + "GROUP BY cc.id, s.id ";
+
+        Query query = entityManager.createQuery(hql);
+        List<Object[]> results = query.getResultList();
+
+        return results.stream()
+                .map(result -> StandardReport.builder()
+                        .criterion(getCertificationCriterion((Long) result[0]))
+                        .standard(getStandard((Long) result[1]))
+                        .count((Long) result[2])
+                        .build())
+                .toList();
+    }
+
+    public List<StandardListingReport> getStandardListingReports() {
+        String hql = "SELECT cc, s, cpd.chplProductNumber "
+                + "FROM CertificationCriterionEntity cc, "
+                + "CertificationEditionEntity ce, "
+                + "CertificationResultEntity cr, "
+                + "CertifiedProductDetailsEntity cpd, "
+                + "CertificationResultStandardEntity crs, "
+                + "StandardEntity s "
+                + "WHERE cc.id = cr.certificationCriterionId "
+                + "AND cc.certificationEditionId = ce.id "
+                + "AND cr.certifiedProductId = cpd.id "
+                + "AND cr.id = crs.certificationResultId "
+                + "AND crs.standard.id = s.id "
+                + "AND cpd.certificationStatusId IN (1,6,7) "
+                + "AND cc.deleted = false "
+                + "AND cr.deleted = false "
+                + "AND crs.deleted = false "
+                + "AND cpd.deleted = false ";
+
+        Query query = entityManager.createQuery(hql);
+        List<Object[]> results = query.getResultList();
+
+        return results.stream()
+                .map(result -> StandardListingReport.builder()
+                        .criterion(((CertificationCriterionEntity) result[0]).toDomain())
+                        .standard(((StandardEntity) result[1]).toDomain())
+                        .chplProductNumber((String) result[2])
+                        .build())
+                .toList();
+    }
+
+    private CertificationCriterion getCertificationCriterion(Long id) {
+        try {
+            return certificationCriterionDao.getById(id);
```

Comment:
I was wondering if you would want to use CertificationCriterionService here which has a method to get by ID. I was just thinking it would be less queries that have to run since we have all the criteria cached in that class. Hibernate has it's own caching thing as well, so I dunno. But I'll throw it out as a suggestion to use that service instead.